### ISSUE TITLE
Booting instance cause infinite recursion in the case of enough user quota and unlimited project quota

### ISSUE THREAD

**Issue by git-harry** [rcbops/rpc-openstack#1278]
###### Please provide the output to 'git describe --all' for  all deployments where you have confirmed it occurs.
liberty-12.2
##### What type of issue is this? Please tick the appropriate box.
- [x] bug - existing functionality that does not work for you as documented
- [ ] feature - new functionality that you would like to see added

##### What is the expected behavior?
https://bugs.launchpad.net/nova/+bug/1491222
https://bugs.launchpad.net/nova/+bug/1591429
##### What is actually happening?
https://bugs.launchpad.net/nova/+bug/1491222
https://bugs.launchpad.net/nova/+bug/1591429
##### What were you attempting to do when you encountered this error (give as much detail as possible)? 
https://bugs.launchpad.net/nova/+bug/1491222
https://bugs.launchpad.net/nova/+bug/1591429
##### How can we reproduce this issue?
https://bugs.launchpad.net/nova/+bug/1491222
https://bugs.launchpad.net/nova/+bug/1591429
##### How can we verify your issue is addressed?
https://bugs.launchpad.net/nova/+bug/1491222
https://bugs.launchpad.net/nova/+bug/1591429
##### What is your evaluation on the importance of this issue, and why?

----------------------------------------------------------------------------------------------------------------------------
###### For each pull request, update the branch line to reference it.
- [ ] master
- [ ] liberty-12.2

##### Upgrade impact. Consider the impact on moving between major, minor and patch releases.
- [ ] This change does not impact upgrades
- [ ] This change addresses a potential upgrade impact
- [ ] The upgrade impact is being tracked by another issue.

The issue tracking the upgrade impact is:

##### Testing impact
- [ ] An issue/pull-request has been create or exists to track the testing gap, and is referenced below
- [ ] This change includes additional testing
- [ ] This change is tested by the existing setup

Additional testing links:

##### Release note
- [ ] This change includes a release note.
- [ ] This change does not require a release note.

##### Does your change require documentation?
- [ ] Yes, and I submitted an issue to [Docs issues](https://github.com/rackerlabs/docs-rpc/issues "Docs issues")
- [ ] Yes, and I contributed a documentation patch to [docs-rpc](https://github.com/rackerlabs/docs-rpc "docs-rpc")
- [ ] No documentation required for this change

Additional documentation links (please add links if you have ticked yes):

**Comment by ameserole** [comment]
@git-harry Looks like @JCallicoat proposed a backport to liberty upstream already [1]. Does this need to be added to RPC-O also? 

[1] https://review.openstack.org/#/c/329231/

**Comment by git-harry** [comment]
@ameserole I don't understand your question.
I see that review has been there for 5 weeks. We need to be working with the community to try and get it merged given the importance support place on its resolution.

**Comment by ameserole** [comment]
UPDATE:
The backport is not passing gating tests. I am investigating on how to fix the issues now.

**Comment by b3rnard0** [comment]
@git-harry This was assigned to Andrew but he is going and I must have missed this one at the replenishment meeting. We need someone to continue this work. cc: @mancdaz

**Comment by sigmavirus24** [comment]
@alextricity25 aren't you working on this?

**Comment by alextricity25** [comment]
@sigmavirus24 no. It should be moved back to approved IMO

**Comment by mancdaz** [comment]
@alextricity25 I'll ask differently. Can you work on this one please :)

We can't move work backwards as it creates flow debt.

**Comment by git-harry** [comment]
Those assigned to work on revision releases need to organise around the issues that have lost an owner due to changing team membership. As a group we have committed to this work and so need to do our best to complete it.

**Comment by Frank-ZhangXin** [comment]
@git-harry I saw someone already working on Liberty fix, so should I help him reviewing or open another patch review to fix liberty issue?

**Comment by tomc603** [comment]
I discussed the possibility of @Frank-ZhangXin contributing to the upstream Nova project by helping get this patch merged so we can file an OSA bug for a SHA bump to include it, so we can bump the OSA sha in RPC-O and get this deployed. The change looks small and simple, so barring any serious objections by Nova, I can't see any reason this would be too big a deal.

But, the status of this bug should be "blocked" until the upstream bug(s) are merged, I would think.

**Comment by git-harry** [comment]
@tomc603 the SHA updates are done as part of the RPCO and OSA release processes. If that is all that's left to do, the issue can be moved to pending-SHA-update. If the Nova patch just needs reviews, this issue can be moved to needs-review or if @Frank-ZhangXin is helping create the patch it should stay in doing. The issue shouldn't be blocked if the upstream work needs completing because part of us pulling the issue in is so that we assist with it.

I'll remove the blocked label but if I've missed something let me know and we can change it.

**Comment by Frank-ZhangXin** [comment]
@JCallicoat hi, Jordan. I would like to assist you on the existing backport. Or I am able to take over it if your schedule too packed. What do you think? Thanks

**Comment by Frank-ZhangXin** [comment]
Update: rebase to master, gate test - devstack vm gate test failed, which is caused by timeout of devstack-vm-gate.sh . And other nova network tests failed also. Previous one test related to nova network passed this time. My conclusion is no idea what happened, and need more advise.

**Comment by git-harry** [comment]
@tomc603 is this something you can help Frank with?

**Comment by b3rnard0** [comment]
Currently blocked until we get support feedback.

**Comment by Frank-ZhangXin** [comment]
Update: I've have reached master patch author Rui Chen, and exchange each other's mind. The conclusion so far still is we're getting unrelated gate errors. The patch works for a better nova db unit test rather than triggering devstack and xenserver failed.

**Comment by sigmavirus24** [comment]
I took a look at review. I suspect we'll have to keep rechecking that xenserver CI job until it passes. If it fails a bunch more, we should try to get in touch with the CI maintainers about it.

**Comment by Frank-ZhangXin** [comment]
Update: Citrix XenServer CI -dsvm-tempest-nova-network is failed which is caused by "Invalid disk format 'qcow2' for image.".

**Comment by JCallicoat** [comment]
This is blocked upstream waiting for nova stable brach maintainers to merge the backport. Don't think you need anything from @rcbops/support-feedback.

**Comment by claco** [comment]
@tbreeds Given the time frames around liberty, can you advise on the possibility of this going into liberty upstream? It's something that bubbled up for us and some customers in support.

@jhesketh Can you possibly have a look at this for us from the ci side? The Xenserver CI job seems cranky, and seems like it should be unrelated to the patch, but I don't think we know for sure. Any help would be appreciated, or if you can direct us to the right folk.

**Comment by Frank-ZhangXin** [comment]
@claco thanks for reference

**Comment by tbreeds** [comment]
I'll look at the patch today, with a 30second look I think it'll be okay for liberty.

XenServer is 3rd  party CI so it's more of a nova issues that an Infra issue.  I'll look at that as we may have introduced a regression while that CI was offline.  Also IIUC it isn't blocking the change even with a -1 from that CI we can merge it

**Comment by tbreeds** [comment]
Yeah so the XenServ CI is misconfigured:  http://dd6b71949550285df7dc-dda4e480e005aaa13ec303551d2d8155.r49.cf1.rackcdn.com/31/329231/3/check/dsvm-tempest-nova-network/b323a09/logs/etc/glance/glance-api.conf.txt.gz

Doesn't list qcow2 it has "ami,ari,aki,vhd,raw,iso" so it'll will never pass.  I'll look at adding a fix to tempest to detect this and skip if required

**Comment by tbreeds** [comment]
Actually this is happening because of I41c09d5a682cdb068b97b4bd1be3403ee7973c7c which is relatively new.  We need to get XenServer CI reconfigured

Until then it'll fail everything

**Comment by tbreeds** [comment]
I'll reach out to the CI in question.  This *should* be failing on master as well, but like I said it's not on the critical path

**Comment by tbreeds** [comment]
I've done the needfull.  I'll poke some of the (non-racker) nova-stable-cores in the next few days

**Comment by jhesketh** [comment]
I agree with Tony's assessment here. It's up to the 3rd party CI to fix their tests. Tony let me know if you need any help reaching out to them if you haven't already.

**Comment by Frank-ZhangXin** [comment]
@tbreeds @jhesketh Thanks, guys. Appreciate professional insight and review.

@git-harry @b3rnard0 So this should be marked blocked until upstream CI test fixed in future. Tony has already state the cause very specifically which points out 'qcow2' format missing in nova config.

**Comment by tbreeds** [comment]
Thanks @jhesketh I reached out to them yesterday.

The XenServer CI team has "fixed" the problem by disabling that test :( rather than doing as I suggested and add the appropriate devstack config :(

So this is currently blocked on a second (non-racker) core review of the change in question.

**Comment by Frank-ZhangXin** [comment]
Update:
gate-tempest-dsvm-neutron-multinode-full and Mellanox CI are failed, but don't block patch merging.
@tbreeds Tony do you family any Nova community or IRC channel that we could send our patch to review? Thanks.

**Comment by Frank-ZhangXin** [comment]
Update:
Just got another +2, patched merged.

**Comment by mancdaz** [comment]
SHA bump has been completed and includes the upstream fix for this in liberty https://github.com/rcbops/rpc-openstack/pull/1422
